cleanup, refactor and add missing  self._dde_suppressed checks

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #152657

so two things other than cleanups and refactoring 
1) do not use propagate_real_tensors to resolve eval under guard_or_true/guard_or_false . 
2) do not guard for dimensions of type  DimDynamic.OBLIVIOUS_SIZE under guard_or_true/guard_or_false . 


cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @chenyang78 @kadeng @chauhang @amjames

laithsakka committed:
cleanup, refactor and add missin  self._dde_suppressed checks

[ghstack-poisoned]

laithsakka commented on torch/fx/experimental/symbolic_shapes.py:
```diff
@@ -7194,25 +7209,25 @@ def compute_concrete_val() -> sympy.Basic:
-                            )
-                            self.evaluate_expr(s, forcing_spec=True)
+
+        if not self._suppress_guards_tls():
```
no need for else because except always raise !

laithsakka replied:
ah this will raise a merge conflict i also removed that else int this PR. lets who lands first :p

Skylion007 commented on torch/fx/experimental/symbolic_shapes.py:
```diff
@@ -6841,6 +6841,81 @@ def evaluate_sym_node(
+
+            cur = bisect_left(insts, frame.f_lasti, key=lambda x: x.offset)
+        else:
+            # For Pyhton <= 3.10, instructions are always 2 bytes.
```
Typo Python

laithsakka committed:
Update on "cleanup, refactor and add missing  self._dde_suppressed checks"


so two things other than cleanups and refactoring 
1) do not use propagate_real_tensors to resolve eval under guard_or_true/guard_or_false . 
2) do not guard for dimensions of type  DimDynamic.OBLIVIOUS_SIZE under guard_or_true/guard_or_false . 


cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

pianpwk reviewed (commented):
The refactor makes a lot of sense, but I'm confused by the usage of the `_dde_suppressed` flag? It seems like we would not fallback to oblivious/unbacked hints if `_dde_suppressed = True`, how do these relate?

laithsakka commented:
@pytorchbot merge

huydhn commented:
@pytorchbot revert -m 'Sorry for reverting your change but it seems to cause a test to fail in trunk' -c nosignal

export/test_draft_export.py::TestDraftExport::test_unbacked_div_mod_replacement [GH job link](https://github.com/pytorch/pytorch/actions/runs/14852265833/job/41698389661) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/784c666cae00f85ecf675298ddb056bebaf32f55)

laithsakka commented:
@pytorchbot rebase

laithsakka committed:
Update

[ghstack-poisoned]

laithsakka committed:
Update on "cleanup, refactor and add missing  self._dde_suppressed checks"


so two things other than cleanups and refactoring 
1) do not use propagate_real_tensors to resolve eval under guard_or_true/guard_or_false . 
2) do not guard for dimensions of type  DimDynamic.OBLIVIOUS_SIZE under guard_or_true/guard_or_false . 


cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

laithsakka committed:
Update on "cleanup, refactor and add missing  self._dde_suppressed checks"


so two things other than cleanups and refactoring 
1) do not use propagate_real_tensors to resolve eval under guard_or_true/guard_or_false . 
2) do not guard for dimensions of type  DimDynamic.OBLIVIOUS_SIZE under guard_or_true/guard_or_false . 


cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented:
rebase

laithsakka committed:
Update on "cleanup, refactor and add missing  self._dde_suppressed checks"


so two things other than cleanups and refactoring 
1) do not use propagate_real_tensors to resolve eval under guard_or_true/guard_or_false . 
2) do not guard for dimensions of type  DimDynamic.OBLIVIOUS_SIZE under guard_or_true/guard_or_false . 


cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

laithsakka commented:
@pytorchbot merge -i not relatd

laithsakka commented:
@pytorchbot -i

laithsakka commented:
@pytorchbot merge

laithsakka commented:
@pytorchbot merge -f

laithsakka commented:
@pytorchbot merge -f (i tried to merge one million time)

laithsakka commented:
@pytorchbot merge -f "no faluire merge failing"

malfet commented:
@pytorchbot revert -m "Broke lint" -c nosignal

malfet commented:
> @pytorchbot merge -f "no faluire merge failing"

Not really, lint job was not scheduled for the PR (due to github glitch I guess) and so merge was not happening. You force merged the change, that caused regression in OSS

laithsakka commented:
> > @pytorchbot merge -f "no faluire merge failing"
> 
> Not really, lint job was not scheduled for the PR (due to github glitch I guess) and so merge was not happening. You force merged the change, that caused regression in OSS

what tests are failing? I do not see failing tests on my PR

malfet commented:
> > > @pytorchbot merge -f "no faluire merge failing"
> > 
> > 
> > Not really, lint job was not scheduled for the PR (due to github glitch I guess) and so merge was not happening. You force merged the change, that caused regression in OSS
> 
> what tests are failing? I do not see failing tests on my PR

https://hud.pytorch.org/hud/pytorch/pytorch/1748fa529a93f4aa7b69998ee5387bc91b653502/1?per_page=50&name_filter=Lint%20%2F%20lintrunner-noclang%20%2F%20linux-jo&mergeEphemeralLF=true

But this test was not scheduled on your PR (and that's the reason why merge job was timing out)

laithsakka committed:
Update on "cleanup, refactor and add missing  self._dde_suppressed checks"


so two things other than cleanups and refactoring 
1) do not use propagate_real_tensors to resolve eval under guard_or_true/guard_or_false . 
2) do not guard for dimensions of type  DimDynamic.OBLIVIOUS_SIZE under guard_or_true/guard_or_false . 


cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge -f "updated only lint error"
